src/pages/Swap/index.tsx

```diff
@@ -714,8 +720,8 @@ export default function Swap() {
                                   </Trans>
                                 )}
                               </span>
-                              {approvalState === ApprovalState.PENDING ? (
-                                <Loader stroke="white" />
+                              {approvalPending || approvalState === ApprovalState.PENDING ? (
```

Comment:
is approvalState === PENDING still necessary? can it ever be true while approvalPending is false?

Reply:
I think it can if you continue to use the UX while waiting for the pending approval to go through